### PR TITLE
WIP: Adding Android support

### PR DESCRIPTION
**Pull request by eyal-sasson**
This goes together with [capacitor-ssh-plugin/9](tuzig/capacitor-ssh-plugin/pull/9).
There are actual changes to the source files that I'm not sure are correct but that was the only way to make it work

### CODE REVIEW

**Comment by daonb** [comment]
Which files are you not sure of?

**Comment by eyal-sasson** [comment]
> Which files are you not sure of?

I changed some stuff in `terminal7.js` and `ssh_session.ts` that I'm not sure is necessary:
The app seemed to always check the network status and update so I had to remove the lines doing it.
Also I made changes to how the app handles requests from webexec.

**Comment by eyal-sasson** [comment]
Here's a problem: https://bugs.chromium.org/p/chromium/issues/detail?id=118639
Not sure what to do, we have to find a way to handle key presses.

**Comment by eyal-sasson** [comment]
I'm really not sure what to do - it seems xterm.js uses an `input` event but it lacks some data we need and use.

**Comment by eyal-sasson** [comment]
It might be a virtual keyboard only problem. Using a tool like [scrcpy](https://github.com/Genymobile/scrcpy) to simulate a physical keyboard, everything works fine.

File: android/.gitignore [hunk 1]
```diff
@@ -94,3 +94,8 @@ capacitor-cordova-android-plugins
 
 # Copied web assets
 app/src/main/assets/public
+
+# Generated Config files
+app/src/main/assets/capacitor.config.json
+app/src/main/assets/capacitor.plugins.json
```
Comment (daonb) [inline android/.gitignore:100]: They are already in the repo and why do we have generated config under `/app/src/...` ?

File: src/map.ts [hunk 2]
```diff
@@ -60,6 +60,7 @@ export class T7Map {
                 this.shell.start()
                 resolve()
             })
+
```
Comment (daonb) [inline src/map.ts:63]: Please remove

File: src/ssh_session.ts [hunk 3]
```diff
@@ -218,7 +218,7 @@ export class HybridSession extends SSHSession {
             }
             this.candidate += line
             // ignore echo
-            if (this.sentMessages.indexOf(this.candidate) != -1) {
+            if (this.sentMessages.indexOf(this.candidate) != -1 || !this.candidate.startsWith("{")) {
```
Comment (daonb) [inline src/ssh_session.ts:221]: The line is way too long

File: src/terminal7.js [hunk 4]
```diff
@@ -183,8 +183,8 @@ export class Terminal7 {
                 })
         // hide the modal on xmark click
         // Handle network events for the indicator
-        Network.addListener('networkStatusChange', s => 
-            this.updateNetworkStatus(s))
+        /* Network.addListener('networkStatusChange', s => 
+            this.updateNetworkStatus(s)) */
```
Comment (daonb) [inline src/terminal7.js:187]: why?

**Review by daonb — CHANGES_REQUESTED** [review]
I want to merge it in but there are still a few issues

File: android/.gitignore (hunk 1 above)
Comment (eyal-sasson) [reply android/.gitignore:100]: These were generated by Android Studio when I first built the app, I'm not sure they're needed (the `capacitor.config.json` is the same as the one we have in the root directory.

File: src/terminal7.js (hunk 4 above)
Comment (eyal-sasson) [reply src/terminal7.js:182]: This line causes the app to constantly refresh, I don't know why the event is being called.

Comment (daonb) [reply src/terminal7.js:182]: It''s used to identify when the network is down

Comment (eyal-sasson) [reply src/terminal7.js:182]: image
This is what happens when this check is uncommented (the host is online).

Comment (daonb) [reply src/terminal7.js:182]: We need to improves what happens on network down, but that's a separate issue. 
For now try to protect it against reentrancy. Try adding a `lastNetworkState` and call the handler nly when it's a state change.

**Comment by eyal-sasson** [comment]
I believe it is ready to be merged, though the space key still doesnt work (might be a problem with my keyboard only)
Maybe we should make an issue to split the key events to `onData` and a normal keyListener.

**Pull request merged**